src/vse_sync_pp/analyzers/ppsdpll.py

```diff
@@ -0,0 +1,209 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+import pandas as pd
+import numpy as np
+import allantools
+
+from collections import namedtuple
+from decimal import Decimal
+
+class Analyzer():
+    def __init__(self):
+        self._col1_name = 'timestamp'
+        self._col2_name = 'eecstate'
+        self._col3_name = 'phasestate'
+        self._col4_name = 'phaseoffset'
+        self._data  = pd.DataFrame(columns=['timestamp','eecstate','phasestate','phaseoffset']) 
+    def append(self, data):
+        """Append x and y data points extracted from `data`"""
+        col1_val = getattr(data, self._col1_name)
+        col2_val = getattr(data, self._col2_name)
+        col3_val = getattr(data, self._col3_name)
+        col4_val = getattr(data, self._col4_name)
+        self._data.loc[len(self._data)]=[col1_val,col2_val,col3_val,float(col4_val)]
+    def retrieve(self):
+        """Return the data in Data Frame format"""
+        return self._data
+    def to_visualize_stage(self, col1, col2, col3, col4):
+        elems = (self._col1_name, self._col2_name, self._col3_name, self._col4_name)
+        data = namedtuple('Analyzed', elems)
+        timestamp = Decimal(col1)
+        eecstate = int(col2)
+        phasestate = int(col3)
+        phaseoffset = float(col4)
+        return data(timestamp, eecstate, phasestate, phaseoffset)
+    def transform(self):
+        """Generator to convert from Data Frame to NamedTuple (canonical)"""
+        result = [self.to_visualize_stage(row[0],row[1],row[2],row[3]) for row in zip(self._data['timestamp'],self._data['eecstate'],self._data['phasestate'],self._data['phaseoffset'])]
+        return (result)
+    def calculate(self, samples, prtc_x):
+        """Calculate KPI based on input samples according to certain class prtc_x""" 
+        return samples
+
+class PhaseOffsetTimeError(Analyzer):
+    id_ = 'ppsdpll/phase-offset-time-error'
+    prtc_b = 40
+    prtc_a = 100
+    y_name = "phaseoffset"
+
+    def mask_calculation(self, prtc_x):
+        """
+        Calculate mask
+        Input: mask
+        Output: mask value in ns
+        """
+        match prtc_x:
+            case "PRTC-B":
+                return self.prtc_b
+            case "PRTC-A":
+                return self.prtc_a
+
+    def calculate(self, samples, prtc_x):
+        """
+        Calculate Time Error from PhaseDPLL
+        Input: samples in csv format
+        Output: result
+        """
+        phase_min     = samples.loc[1 : len(samples)].phaseoffset.min()
```

Comment:
I would move `samples.loc[1: len(samples)].phaseoffset` (which I think should be equivalent to `samples.loc[1:].phaseoffset` to a temp var instead of slicing it over and over again.